engine/src/main/java/org/terasology/logic/inventory/ItemPickupAuthoritySystem.java

```diff
@@ -59,20 +66,32 @@ public void onDropItemEvent(DropItemEvent event, EntityRef itemEntity, ItemCompo
         if (!itemEntity.hasComponent(LocationComponent.class)) {
             itemEntity.addComponent(new LocationComponent(event.getPosition()));
         }
+
+        itemEntity.getComponent(ItemComponent.class).pickableUp = false;
+        long timeToPickUp = itemEntity.getComponent(ItemComponent.class).pickUpTime;
+        delayManager.addDelayedAction(itemEntity, DELAYED_PICKUP_ACTION_ID, timeToPickUp);
     }
 
+    @ReceiveEvent
+    public void onDelayedPickup(DelayedActionTriggeredEvent event, EntityRef entityRef) {
```

Comment:
You could possibly add the ItemComponent here as a parameter that way some optimizations can be done by only firing this code when a ItemComponent is present.  Also,  then you dont have to do a hasComponent or null check (which you dont presently have)